libs/journeys/ui/src/components/Step/Step.spec.tsx

```diff
@@ -179,64 +268,65 @@ describe('Step', () => {
     }))
 
     render(
-      <MockedProvider
-        mocks={[
-          {
-            request: {
-              query: STEP_VIEW_EVENT_CREATE,
-              variables: {
-                input: {
-                  id: 'uuid',
-                  blockId: 'Step1',
-                  value: 'Step {{number}}'
-                }
-              }
-            },
-            result
-          }
-        ]}
-      >
-        <JourneyProvider>
-          <Step
-            {...block}
-            wrappers={{
-              Wrapper: ({ children }) => <div>{children}</div>
-            }}
-          />
+      <MockedProvider mocks={[{ ...mockStepViewEventCreate, result }]}>
+        <JourneyProvider value={{ journey }}>
+          <Step {...block} />
         </JourneyProvider>
       </MockedProvider>
     )
+
     await waitFor(() => expect(result).not.toHaveBeenCalled())
   })
 
-  it('should render blocks', () => {
-    const { getByText } = render(
+  it('should not send stepViewEvent if not activeStep - broken', async () => {
+    mockUuidv4.mockReturnValueOnce('uuid')
+    treeBlocksVar([])
+    blockHistoryVar([block, { ...block, id: 'Step2' }])
+
+    render(
       <MockedProvider>
```

Comment:
Does this work? If not, lets make one just for this test
```suggestion
      <MockedProvider mocks={[mockStepViewEventCreate]}>
```